jdftx/scripts/ase/JDFTx.py

```diff
@@ -197,11 +240,11 @@ def constructInput(self, atoms):
 			inputfile += 'coulomb-truncation-embed %g %g %g\n' % tuple(center.tolist())
 		
 		#Add dump commands
-		inputfile += '\ndump End State Forces Ecomponents'
+		inputfile += "".join(["dump %s %s\n" % (when, what) for when, what in self.dumps])
 
 		# Cache this calculation to history
-		self.lastAtoms = copy.deepcopy(atoms)
-		self.lastInput = copy.deepcopy(self.input)
+		self.lastAtoms = atoms.copy()
```

Comment:
Ran into a weird recursion error when using `deepcopy`, so switched to the ASE method.